tests/test_exceptions.py

```diff
@@ -1,15 +1,62 @@
-from ocpp.exceptions import ProtocolError
+import pytest
+
+from ocpp.exceptions import (
+    ProtocolError,
+    TypeConstraintViolationError,
+    FormatViolationError,
+)
+from ocpp.messages import validate_payload, Call
 
 
 def test_exception_with_error_details():
-    exception = ProtocolError("Some error", {'key': 'value'})
+    exception = ProtocolError("Some error", {"key": "value"})
 
     assert exception.description == "Some error"
-    assert exception.details == {'key': 'value'}
+    assert exception.details == {"key": "value"}
 
 
 def test_exception_without_error_details():
     exception = ProtocolError()
 
     assert exception.description == "Payload for Action is incomplete"
     assert exception.details == {}
+
+
+def test_exception_show_triggered_message_type_constraint():
+    """chargePointVendor should be a string, not an integer, so this should
+    raise a TypeConstraintViolationError
+    """
+    call = Call(
+        unique_id=123456,
+        action="BootNotification",
+        payload={"chargePointVendor": 1,
+                 "chargePointModel": "SingleSocketCharger"},
+    )
+
+    regex_pattern = (
+        r"'ocpp_message': <Call - unique_id=123456, action=BootNotification, "
+        r"payload={'chargePointVendor': 1, "
+        r"'chargePointModel': 'SingleSocketCharger'}"
+    )
+
+    with pytest.raises(TypeConstraintViolationError, match=regex_pattern):
+        validate_payload(call, "1.6")
```

Comment:
Nitpick:

Is is, technically, a regex pattern. But as far as I can see it is a trivial one. I would rename it to `pattern`, because then reviewers don't start to look for (real) regex patterns.

Since parsing regular expressions is time consuming, you could just use `pytest.raises` like this:
https://docs.pytest.org/en/7.1.x/how-to/assert.html#assertions-about-expected-exceptions

Both suggestions are completely optional and more a matter of taste.